test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlDocumentTest/HandlesNewLineInNestedBlock.stree.txt

```diff
@@ -1,33 +1,37 @@
-Markup block - Gen<None> - 29 - (0:0,0)
-    Markup span - Gen<Markup> - [] - SpanEditHandler;Accepts:Any - (0:0,0) - Tokens:1
-        SyntaxKind.Unknown;[];
-    Statement block - Gen<None> - 21 - (0:0,0)
-        Transition span - Gen<None> - [@] - SpanEditHandler;Accepts:None - (0:0,0) - Tokens:1
-            SyntaxKind.Transition;[@];
-        MetaCode span - Gen<None> - [{] - SpanEditHandler;Accepts:None - (1:0,1) - Tokens:1
-            SyntaxKind.LeftBrace;[{];
-        Code span - Gen<Stmt> - [LF] - AutoCompleteEditHandler;Accepts:Any,AutoComplete:[<null>];AtEOL - (2:0,2) - Tokens:1
+SyntaxKind.HtmlDocument - [0..29) - FullWidth: 29 - Slots: 1 - [@{LF@if(true){LF} LF}LF<html>]
+    SyntaxKind.HtmlMarkupBlock - [0..29) - FullWidth: 29 - Slots: 1
+        SyntaxKind.HtmlTextLiteral - [0..0) - FullWidth: 0 - Slots: 1 - Gen<Markup> - SpanEditHandler;Accepts:Any
+            SyntaxKind.Unknown;[];
+        SyntaxKind.CSharpCodeBlock - [0..21) - FullWidth: 21 - Slots: 1
+            SyntaxKind.CSharpStatement - [0..21) - FullWidth: 21 - Slots: 2
+                SyntaxKind.CSharpTransition - [0..1) - FullWidth: 1 - Slots: 1 - Gen<None> - SpanEditHandler;Accepts:None
+                    SyntaxKind.Transition;[@];
+                SyntaxKind.CSharpStatementBody - [1..21) - FullWidth: 20 - Slots: 3
+                    SyntaxKind.RazorMetaCode - [1..2) - FullWidth: 1 - Slots: 1 - Gen<None> - SpanEditHandler;Accepts:None
+                        SyntaxKind.LeftBrace;[{];
+                    SyntaxKind.CSharpCodeBlock - [2..20) - FullWidth: 18 - Slots: 1
+                        SyntaxKind.CSharpStatementLiteral - [2..4) - FullWidth: 2 - Slots: 1 - Gen<Stmt> - AutoCompleteEditHandler;Accepts:Any,AutoComplete:[<null>];AtEOL
```

Comment:
Okay that makes sense. This is a good time to bring up two other literal types I've defined,
`CSharpHiddenLiteralSyntax` - to represent the text that we don't want to render in the output (`SpanChunkGenerator.Null` in the old world)
`CSharpNoneLiteralSyntax` - to represent the text where we don't want any design time intellisense (`SpanKind.None` in the old world)
Thoughts on those?

Reply:
> CSharpNoneLiteralSyntax

Hmm, is it prefixed with `CSharp` because it contains `CSharp` tokens? Makes me think we should have an unclassified token type / span kind. Basically if the node was named `UnclassifiedTextLiteralSyntax` with unclassified tokens it'd truly represent what was intended.

Reply:
As for `CSharpHiddenLiteralSyntax`, i'm uncertain. What's an example of a `SpankChunkGenerator.Null` again? Having a hard time remembering where we do that.

Reply:
Missed these questions earlier,

> Hmm, is it prefixed with CSharp because it contains CSharp tokens?

We currently use it in only one place. Whitespace at the end of SingleLine extensible directives. Renaming it to unclassified should work,

> What's an example of a SpankChunkGenerator.Null again?

One example is just like in #2594,
```
@{ @@Da }
   ^
```
That transition is a code span with SpanChunkGenerator.Null